lib/groonga/command/format/elasticsearch.rb

```diff
@@ -57,27 +59,27 @@ def json(options={})
                 }
               end
               components << JSON.generate(header)
+            when :columns
+              value.split(",").each do |column_name|
+                column_names << column_name
+              end
             when :values
               record = JSON.parse(value)
 
               if record[0].is_a?(::Array)
-                column_names = nil
-
                 record.each do |load_value|
-                  if column_names.nil?
-                    column_names = load_value
-                    next
-                  end
                   body = {}
                   column_values = load_value
                   column_names.zip(column_values) do |column_name, column_value|
                     body[column_name] = column_value
                   end
                   components << JSON.generate(body)
+                  components << JSON.generate(header) if load_value != record.last
                 end
               else
                 record.each do |load_value|
                   components << JSON.generate(load_value)
+                  components << JSON.generate(header) if load_value != record.last
```

Comment:
This is weird.
If there is no "values", can this code generate correct output?

```ruby
      class Elasticsearch
        def initialize(command)
          @command = command
        end

        def json(options={})
          return nil unless @command.is_a?(Load)

          components = []
          action = build_action
          each_source do |source|
            components << JSON.generate(action)
            components << JSON.generate(source)
          end
          components.join("\n")
        end

        private
        def elasticsearch_version
          options[:version] || 5
        end

        def build_action
          index_name = @command.table.downcase
          case elasticsearch_version
          when 5, 6
            {
              "index" => {
                "_index" => index_name,
                "_type" => "groonga",
              }
            }
          when 7
            {
              "index" => {
                "_index" => index_name,
                "_type"=>"_doc",
              }
            }
          else
            {
              "index" => {
                "_index" => index_name,
              }
            }
          end
        end

        def each_source
          columns = @command.columns
          values = @command.values || []
          values.each do |value|
            if value.is_a?(::Array)
              if columns.nil?
                columns = value
              else
                yield(Hash[columns.zip(value)])
              end
            else
              yield(value)
            end
          end
        end
```

Reply:
Thank you for your suggestion.

I've changed your suggestion code a little bit.
Because `@command.columns` and `@command.values` are nil, I used `@command[:columns]` and `@command[:values]` instead.